cmake/compiler/arcmwdt/generic.cmake

```diff
@@ -2,6 +2,12 @@
 
 # Configures CMake for using ccac
 
+find_program(CMAKE_DTS_PREPROCESS_C_COMPILER arc-elf32-gcc)
+
+if(CMAKE_DTS_PREPROCESS_C_COMPILER STREQUAL CMAKE_DTS_PREPROCESS_C_COMPILER-NOTFOUND)
```

Comment:
Can be simplified:
```suggestion
if(CMAKE_DTS_COMPILER)
```

> if(<constant>)
>
>    False if the constant is 0, OFF,..., or ends in the suffix -NOTFOUND.

Reply:
Ok, I'll fix the check, however I'll use CMAKE_DTS_PREPROCESS_C_COMPILER variable name as we already have 'dtc' which is dts compiler (so CMAKE_DTS_COMPILER name will be confusing)

Reply:
accepted, `CMAKE_DTS_PREPROCESSOR` is fine.

Reply:
done